x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/modelsnapshots/RestUpdateModelSnapshotAction.java

```diff
@@ -55,10 +55,6 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
             parser
         );
 
-        return channel -> client.execute(
-            UpdateModelSnapshotAction.INSTANCE,
-            updateModelSnapshot,
-            new RestStatusToXContentListener<>(channel)
-        );
+        return channel -> client.execute(UpdateModelSnapshotAction.INSTANCE, updateModelSnapshot, new RestToXContentListener<>(channel));
```

Comment:
This uses UpdateModelSnapshotAction.Response, which was always returning RestStatus.OK